### PR TITLE
Enable discriminator/kernel return in QASM and add defaults

### PR DESCRIPTION
**Pull request by zachschoenfeld33**
### Summary
This PR updates the backend configuration schema such that qasm will return `kernels/discriminators` if supported. The default backend kernel and discriminator will also be returned.

This is done in a backwards compatible manner so that if the qasm configuration does not return these properties, no error will occur.

### CODE REVIEW

File: qiskit/test/mock/fake_openpulse_2q.py [hunk 1]
```diff
@@ -58,6 +58,8 @@ def __init__(self):
             ],
             meas_kernels=['kernel1'],
             discriminators=['max_1Q_fidelity'],
+            default_meas_kernel='kernel1',
```
Comment (taalexander) [inline qiskit/test/mock/fake_openpulse_2q.py:61]: might as well make these `hw_boxcar` and `linear_discriminator`.

File: qiskit/test/mock/fake_qasm_simulator.py [hunk 2]
```diff
@@ -34,7 +34,11 @@ def __init__(self):
             open_pulse=False,
             memory=True,
             max_shots=65536,
-            gates=[GateConfig(name='TODO', parameters=[], qasm_def='TODO')]
+            gates=[GateConfig(name='TODO', parameters=[], qasm_def='TODO')],
+            meas_kernels=['boxcar'],
+            discriminators=['linear', 'quadratic'],
```
Comment (taalexander) [inline qiskit/test/mock/fake_qasm_simulator.py:39]:
```suggestion
            discriminators=['linear_discriminator', 'quadratic_discriminator'],
```

File: qiskit/test/mock/fake_qasm_simulator.py [hunk 3]
```diff
@@ -34,7 +34,11 @@ def __init__(self):
             open_pulse=False,
             memory=True,
             max_shots=65536,
-            gates=[GateConfig(name='TODO', parameters=[], qasm_def='TODO')]
+            gates=[GateConfig(name='TODO', parameters=[], qasm_def='TODO')],
+            meas_kernels=['boxcar'],
```
Comment (taalexander) [inline qiskit/test/mock/fake_qasm_simulator.py:38]:
```suggestion
            meas_kernels=['hw_boxcar'],
```

File: qiskit/test/mock/fake_qasm_simulator.py [hunk 4]
```diff
@@ -34,7 +34,11 @@ def __init__(self):
...
+            discriminators=['linear', 'quadratic'],
+            default_meas_kernel='boxcar',
```
Comment (taalexander) [inline qiskit/test/mock/fake_qasm_simulator.py:40]:
```suggestion
            default_meas_kernel='hw_boxcar',
```

File: qiskit/test/mock/fake_qasm_simulator.py [hunk 5]
```diff
@@ -34,7 +34,11 @@ def __init__(self):
...
+            default_discriminator='linear'
```
Comment (taalexander) [inline qiskit/test/mock/fake_qasm_simulator.py:41]:
```suggestion
            default_discriminator='linear_discriminator'
```

**Review by taalexander — COMMENTED** [review]
We still need to decide if it best to add these defaults to the configuration.

File: qiskit/test/mock/fake_openpulse_2q.py (hunk 1 above)
Comment (zachschoenfeld33) [reply qiskit/test/mock/fake_openpulse_2q.py:61]: These are tests, so it really doesn't matter. And I don't want to change it if we change some of the default discriminators. But I will change it for now.

Comment (zachschoenfeld33) [reply qiskit/test/mock/fake_openpulse_2q.py]: Addressed in https://github.com/Qiskit/qiskit-terra/pull/5131/commits/2bb40181110801931b8c77e7e114be2cfeeeca9f.

File: qiskit/test/mock/fake_qasm_simulator.py (hunk 3 above)
Comment (zachschoenfeld33) [reply qiskit/test/mock/fake_qasm_simulator.py]: Addressed in https://github.com/Qiskit/qiskit-terra/pull/5131/commits/2bb40181110801931b8c77e7e114be2cfeeeca9f.

File: qiskit/test/mock/fake_qasm_simulator.py (hunk 5 above)
Comment (zachschoenfeld33) [reply qiskit/test/mock/fake_qasm_simulator.py]: Addressed in https://github.com/Qiskit/qiskit-terra/pull/5131/commits/2bb40181110801931b8c77e7e114be2cfeeeca9f.

File: qiskit/test/mock/fake_qasm_simulator.py (hunk 4 above)
Comment (zachschoenfeld33) [reply qiskit/test/mock/fake_qasm_simulator.py]: Addressed in https://github.com/Qiskit/qiskit-terra/pull/5131/commits/2bb40181110801931b8c77e7e114be2cfeeeca9f.

File: qiskit/test/mock/fake_qasm_simulator.py (hunk 2 above)
Comment (zachschoenfeld33) [reply qiskit/test/mock/fake_qasm_simulator.py]: Addressed in https://github.com/Qiskit/qiskit-terra/pull/5131/commits/2bb40181110801931b8c77e7e114be2cfeeeca9f.

File: qiskit/providers/models/backendconfiguration.py [hunk 6]
```diff
@@ -497,8 +523,8 @@ def __init__(self,
             dt: Qubit drive channel timestep in nanoseconds.
             dtm: Measurement drive channel timestep in nanoseconds.
             rep_times: Supported repetition times (program execution time) for backend in μs.
-            meas_kernels: Supported measurement kernels.
-            discriminators: Supported discriminators.
+            meas_kernels (List[str]): Available measurement kernels on the backend.
```
Comment (taalexander) [inline qiskit/providers/models/backendconfiguration.py:526]: These should now be set by calling the super class.

**Review by taalexander — CHANGES_REQUESTED** [review]

**Comment by jakelishman** [comment]
Closing as stale.

**Pull request closed without merging**